src/st/no3/4CA54.c

```diff
@@ -0,0 +1,1068 @@
+                        ent_s0->palette = self->hitEffect;
+                    } else {
+                        ent_s0->palette = self->palette;
+                    }
+                }
+            }
+            break;
+        case 1:
+            AnimateEntity(&D_80182900, self);
+            self->ext.factory.unk80 += 1;
+            self->unk6C -= 2;
+            if (self->unk6C == 0x40) {
+                PlaySfxPositional(0x780);
+            }
+            if (!self->unk6C) {
+                self->drawMode = DRAW_UNK_40 | DRAW_TPAGE;
+                self->palette = 0x15F;
+                self->unk6C = 0x80;
+                self->step_s += 1;
+            }
+            break;
+        case 2:
+            AnimateEntity(&D_80182900, self);
+            self->unk6C -= 2;
+            if (!self->unk6C) {
+                DestroyEntity(self);
+                DestroyEntity(self + 1);
+                return;
+            }
+            break;
+        }
+        break;
+    case 12:
+        ent_s0 = self + 2;
+        ent_s4 = self + 3;
+        switch (self->step_s) {
+        case 0:
+            ent_s0->animFrameDuration = 0;
+            ent_s0->animFrameIdx = 0;
+            ent_s0->ext.strongWarg.unk7C = 1;
+
+            if (!(Random() & 7)) {
+                AnimateEntity(&D_8018296C, ent_s0);
+                AnimateEntity(&D_80182980, self);
+                self->step_s += 1;
+            } else {
+                AnimateEntity(&D_80182990, ent_s0);
+                AnimateEntity(&D_801829B4, self);
+                self->step_s += 2;
+            }
+            self->animCurFrame = 0x32;
+            break;
+        case 1:
+            var_s1 = AnimateEntity(&D_8018296C, ent_s0);
+            AnimateEntity(&D_80182980, self);
+            if (self->velocityX != 0) {
+                if (self->velocityX < 0) {
+                    self->velocityX += FIX(0.5);
+                    EntityUnkId14Spawner(
+                        self, 1, 1, 0, 38, ((Random() & 3) + 1), -4);
+                    EntityUnkId14Spawner(
+                        self, 1, 1, 26, 38, ((Random() & 3) + 1), -4);
+                } else {
+                    self->velocityX -= FIX(0.5);
+                    EntityUnkId14Spawner(
+                        self, 1, 1, -26, 38, ((Random() & 3) + 1), 4);
+                    EntityUnkId14Spawner(
+                        self, 1, 1, 0, 38, ((Random() & 3) + 1), 4);
+                }
+            } else {
+                ent_s4->hitboxState = 0;
+                ent_s4->attackElement = self->attackElement;
+                ent_s4->attack = self->attack;
+            }
+
+            if ((var_s1 & 0x80) && (ent_s0->animFrameIdx == 5)) {
+                if (self->facingLeft) {
+                    self->velocityX = FIX(4.0);
+                } else {
+                    self->velocityX = FIX(-4.0);
+                }
+
+                PlaySfxPositional(0x6C7);
```

Comment:
My counting may be wrong but I think we have an enum https://github.com/Xeeynamo/sotn-decomp/blob/c228c6f26483c4970b7f2d0f11a300fa4ee67f56/include/sfx.h#L502